Added robot configuration documentation (#234)
icub-tech-iit/documentation: docs/icub_robot_configuration/icub_robot_configuration_index.md

[**Naming convention**](./naming_convention/naming_convention.md)

 - [Rules for naming devices](./naming_convention/naming_convention.md#1-rules-for-naming-devices)

 - [Rules for naming wrappers](./naming_convention/naming_convention.md#2rules-for-naming-wrapper-file)

 - [Rules for naming joints](./naming_convention/naming_convention.md#3rules-for-naming-joints)

[**Configuration of devices**](./devices_configuration/devices_configuration.md)

 - [Motor control device](./devices_configuration/devices_configuration.md#motor-control-device)

 - [Motor Control on iCub](./devices_configuration/devices_configuration.md#motor-control-on-icub)

 - [FT sensor device](./devices_configuration/devices_configuration.md#ft-sensor-device)

 - [Mais device](./devices_configuration/devices_configuration.md#mais-device)

 - [Skin device](./devices_configuration/devices_configuration.md#skin-device)

 - [Inertial device](./devices_configuration/devices_configuration.md#inertial-device)

[**Configuration of calibrator module**](./calibrator_configuration/calibrator_configuration.md)

> ## Note for robots in iCub Facility (iit Genoa, Italy)
> Usually the robots in iCub Facility run developing software, that is, they use the same software modules, but on the development branch instead of the stable one (master).
> This mean that in the newest version may be available new parameters or there are some changes that are not described in the last stable version. You can find these difference in the changelog.

Reviewer: ```suggestion
## Note for robots hosted at the iCub Tech Facility
Robots hosted at the iCub Tech Facility often run developing software, that is, they use the same software modules, but on the development branch instead of stable (master).
This means that in the newest version new parameters may be available or there might be some changes that are not described in the latest stable version. You can find these differences in the changelog.
```

> If you are interesting in the meaning of this changes, please check iCubTemplates/iCubTemplateVX_0, where X is the last version in the devel branch.

Reviewer: ```suggestion
If you are interested in the rationale underlying these changes, please check the files `[iCubTemplates](https://github.com/robotology/robots-configuration/tree/devel/iCubTemplates)/iCubTemplateVX_0`, where `X` identifies the latest version in the devel branch.
```
